src/dss/remove_notebook.py

```diff
@@ -0,0 +1,34 @@
+from lightkube import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.apps_v1 import Deployment
+from lightkube.resources.core_v1 import Service
+
+from dss.config import DSS_NAMESPACE
+from dss.logger import setup_logger
+
+# Set up logger
+logger = setup_logger("logs/dss.log")
+
+
+def remove_notebook(name: str, lightkube_client: Client) -> None:
+    """
+    Removes a Notebook server from the Kubernetes cluster.
+
+    Args:
+        name (str): The name of the notebook server.
+        lightkube_client (Client): The Kubernetes client.
+    """
+
+    try:
+        lightkube_client.delete(res=Deployment, name=name, namespace=DSS_NAMESPACE)
+        lightkube_client.delete(res=Service, name=name, namespace=DSS_NAMESPACE)
+    except ApiError as err:
+        if err.status.code == 404:
+            logger.warn(
+                "Failed to delete K8s resources not found. Ignoring remove-notebook.\nTry dss list-notebooks to use the correct notebook name."  # noqa E501
+            )
+            return
```

Comment:
Should this count as a successful execution of `dss remove-notebook`?  As is, this warns and then completes successfully, so something like:
```
dss remove-notebook notARealNotebook && echo success
```
would show the warning and then echo "success"

imo, we should fail here rather than return and succeed (`exit(1)` maybe? or probably raise a custom exception that the `main.py` catches and `main.py` then does `exit(1)`).  Its a bit arbitrary, but I don't think we want to communicate that things went as we hoped.

For comparison, doing `kubectl delete namespace not-real` exits with error code 1

Reply:
I think this applies to all the dss commands we've implemented so far, what we've done up till now is log errors rather than exit with error code. I will bring this up with the team so we can make a decision.